lib/gameboard/board.rb

```diff
@@ -0,0 +1,495 @@
+    #
+    #   diagonal(true)
+    #     #=> [
+    #          [[0, 0], [1, 1], [2, 2]],
+    #          [[0, 2], [1, 1], [2, 0]],
+    #          [[0, 1], [1, 2]],
+    #          [[0, 1], [1, 0]],
+    #          [[0, 2]],
+    #          [[0, 0]],
+    #          [[1, 0], [2, 1]],
+    #          [[1, 2], [2, 1]],
+    #          [[2, 0]],
+    #          [[2, 2]]
+    #         ]
+    #
+    def diagonal(coords = false)
+      diagonals = []
+
+      height.times do |i|
+        diagonals << get_diagonal([0, i], coords)
+        diagonals << get_diagonal([0, height - 1 - i], coords, false)
+      end
+      (1...width).each do
+        |i| diagonals << get_diagonal([i, 0], coords)
+        diagonals << get_diagonal([i, height - 1], coords, false)
+      end
+
+      diagonals
+    end
+
+    # Public: Find a cell at given coordinate.
+    #
+    # coord - An X,Y coordinate passed as an Array.
+    #
+    # Example
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #
+    #   find_cell([1,0])
+    #     #=> Y
+    #
+    #   find_cell([2,2])
+    #     #=> 3
+    #
+    # find_cell([0,1])
+    #     #=> A
+    #
+    def find_cell(coord)
+      board.find {|cell| cell.coord.position == coord}
+    end
+
+    # Public: Check if the gameboard is full relative to the default cell.
+    #
+    #
+    #
+    # Example
+    #
+    #    Board.new(height: 3, width: 3)
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #   full?
+    #     #=> true
+    #
+    #   Board.new(height: 3, width: 3, cells: "Y")
+    #
+    #   full?
+    #     #=> false (because our default piece is now "Y")
+    #
+    def full?
+      board.none? { |cell| cell.value == cells  }
+    end
+
+    # Public: Return every row on the gameboard.
+    #
+    # coords: An optional boolean to specify returning an Array of coordinates
+    #         instead of values.
+    #
+    # Example
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #
+    #   horizontal
+    #     #=> [
+    #          ["X", "Y", "Z"],
+    #          ["A", "B", "C"],
+    #          [1, 2, 3]
+    #         ]
+    #
+    #   horizontal(true)
+    #     #=> [
+    #          [[0, 0], [1, 0], [2, 0]],
+    #          [[0, 1], [1, 1], [2, 1]],
+    #          [[0, 2], [1, 2], [2, 2]]]
+    #         ]
+    #
+    def horizontal(coords = false)
+      rows = []
+      height.times do |y|
+        rows << board.select { |cell| cell.coord.y == y }.map do |cell|
+          coords ? cell.coord.position : cell.value
```

Comment:
Although you might want to purge this feature entirely, if it does remain, It probably deserves its own method, since it is the most repeated bit of code I've spotted so far.